### PR TITLE
Modify modals for consistency

### LINKED ISSUE #403: Facet "More" dialog has a double header
Two red bars, differently rendered, both with an operable close X
screen shot 2016-01-04 at 10 48 58 am
A bug perhaps, but not one that gets in the way.

### LINKED ISSUE #408: Issues with Open for Modification button
This is on the detail page. The modal opens up fine, but cancel is inert and submitting the form with the Open button gets a "Page not found" error.
screen shot 2016-01-04 at 3 18 06 pm

### LINKED ISSUE #412: Modal error display
Not a bug (beyond something hiccuping in stage) but rather just sharing this image of the modal for a formatted workflow when something -- I'm guessing WF services -- aren't working. Maybe you haven't seen it or would just like to see it in action ...
screen shot 2016-01-05 at 12 17 37 am

### PR DESCRIPTION
**Pull request by mejackreed**
This pull request standardizes the approach to modals in Argo.

Closes #403 
Closes #408 60e94b4
Partially addresses #409 
Closes #412 

What this PR aims to do is use Rails template hierarchy to provide views for modals the the non-modal view of a page. When a view is requested it will render using the path: `[action].[format].erb`. For html format requests, the `[action].html.erb` template will be called, for js format requests (modals) the `[action].js.erb` template will be called. Both of these templates provide containers for the partial `_[action].html.erb` which provides the shared display view.

Modals/pages affected:

### Facet -> more
screen shot 2016-01-06 at 6 07 32 pm

### dor/reindex
Note: Per #415 this will actually go away at some point
screen shot 2016-01-06 at 5 53 11 pm

### items/add_workflow
screen shot 2016-01-06 at 5 56 56 pm

### items/close_version_ui
Note: partially addresses #409, still needs select box options to work
screen shot 2016-01-06 at 5 58 44 pm

### items/collection_ui
screen shot 2016-01-06 at 6 00 21 pm

### items/content_type
screen shot 2016-01-06 at 6 01 21 pm

### items/open_version_ui
screen shot 2016-01-06 at 6 02 05 pm

### items/rights
screen shot 2016-01-06 at 6 02 50 pm

### items/source_id_ui
screen shot 2016-01-06 at 6 03 17 pm

### items/tags_ui
screen shot 2016-01-06 at 6 04 20 pm

### items/workflow_history_view
screen shot 2016-01-06 at 6 05 37 pm

### items/workflow_view
screen shot 2016-01-06 at 6 06 20 pm

### view/:id/dc
screen shot 2016-01-06 at 9 24 15 pm
screen shot 2016-01-06 at 9 24 43 pm

### view/:id/purl_preview
screen shot 2016-01-06 at 9 25 34 pm

### CODE REVIEW

**Comment by LynnMcRae** [comment]
Changes in dev looking mighty fine.  One miss is the formatted workflow modal, shown in in current stage and dev
screen shot 2016-01-05 at 9 27 53 am

screen shot 2016-01-05 at 7 46 51 am

**Comment by mejackreed** [comment]
Thanks for the feedback @LynnMcRae . f0d81791061db7ee537c3191597b6b67679d300f only addresses blue button modals. Additional work being down will address workflow and other modals.

File: app/assets/javascripts/jquery.persistentModal.js [hunk 1]
```diff
@@ -105,20 +98,23 @@
...
           '<div class="modal persistent-modal" tabindex="-1" role="modal" data-persistent-modal-url="' + linkTarget + '" data-backdrop="static" data-keyboard="false">',
-          '  <div class="modal-dialog">',
-          '    <div class="modal-header">',
-          '      <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>',
-          '      <h3 class="modal-title">TITLE</h3>',
-          '    </div>',
+          '  <div class="modal-dialog modal-lg">',
           '    <div class="modal-content">',
+          '      <div class="modal-header">',
+          '        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>',
```
Comment (mejackreed) [inline app/assets/javascripts/jquery.persistentModal.js]: I'd like to leave this the way it is for styling readability, if possible. Also open to other suggestions.

Comment (jmartin-sul) [inline app/assets/javascripts/jquery.persistentModal.js]: definitely agree, i'd disregard hound here for sure.

**Comment by drh-stanford** [comment]
The "View full Dublin Core" and "View MODS" links on the item show page come up with broken modals (no way to close them).


screen shot 2016-01-06 at 3 11 44 pm

screen shot 2016-01-06 at 3 13 18 pm

**Comment by mejackreed** [comment]
Thanks for catching that @drh-stanford . I've updated the PR with the modified partials to work with the new pattern.

**Comment by drh-stanford** [comment]
The modals for the datastream views have a Cancel button instead of an X close in the header. Is that expected?

screen shot 2016-01-07 at 11 10 01 am

**Comment by mejackreed** [comment]
@drh-stanford Thanks for catching that. It was an oversight on my part. I've amended and pushed the last commit.

**Comment by drh-stanford** [comment]
When you click on a workflow from the show page, then click Diagram, the modal is missing its close button.

screen shot 2016-01-07 at 12 01 16 pm
screen shot 2016-01-07 at 12 00 56 pm

**Comment by mejackreed** [comment]
@drh-stanford Thanks for pointing that out. I engaged @LynnMcRae specifically about the use of that feature. It seems that it's used not very much. I couldn't figure out a way to make that one route work with the new modal system. I've created a technical debt issue https://github.com/sul-dlss/argo/issues/430 to address this later down the road.

**Comment by drh-stanford** [comment]
ok, then it all looks good. i've checked all the modals that i can find and that you pointed out above.

**Pull request merged**